saga/saga.go

```diff
@@ -201,23 +194,27 @@ type sagaUpdate struct {
 }
 
 //
-// logs the specified message durably to the SagaLog & updates internal state if its a valid state transition
+// checks the message is a valid transition and logs the specified message durably to the SagaLog
+// if msg is an invalid transition, it will neither log nor update internal state
```

Comment:
I'd add a comment saying that it returns the original state when there's an err since that's not typical behavior in our codebase.